pkg/testing/integration/program.go

```diff
@@ -736,6 +744,14 @@ func newProgramTester(t *testing.T, opts *ProgramTestOptions) *ProgramTester {
 		opts.SkipExportImport = true
 		opts.SkipEmptyPreviewUpdate = true
 	}
+	if opts.RequireService {
```

Comment:
Bits of new magic in here? It will now default to filestate unless the user sets flags? And what about the skip behavior this is new?

Reply:
New magic, yeah. This should make the local dev experience when running `make` smoother, as tests will run against filestate if possible, and skip if not.

In our CI harness, we always set `PULUMI_ACCESS_TOKEN`, so no skipping in CI.

Reply:
Ah nice, maybe worth a quick in-source comment.